Assignment2Q2.java

```diff
@@ -0,0 +1,112 @@
+import java.util.LinkedList;
+
+/**
+ * Created by Alexandra on 23.05.2017.
+ */
+public class Assignment2Q2 {
+
+    public static class TreeNode <T> {

```

Comment:
As you are using this class in both questions you could put it in a separate file. And then depend on it from each question file.